### PR TITLE
116-function-to-take-config-and-produce-pv-events-grouped-by-job-name

### PR DESCRIPTION
**Pull request by AdrianMontaguSmartDCSIT**
This PR adds a function otel_to_pv that takes in the config, streams data from the data holder and converts OTelEvents to PVEvents.

### CODE REVIEW

File: tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py [hunk 1]
```diff
@@ -602,3 +603,155 @@ def mock_fetch_data_holder(config: IngestDataConfig) -> SQLDataHolder:
...
+        )
+        monkeypatch.setattr(
+            "tel2puml.otel_to_pv.sequence_otel_v2.fetch_data_holder",
+            mock_fetch_data_holder,
+        )
+
+        result = otel_to_pv(ingest_data_config)
+
+        events = []
+        valid_job_names = {"test_name"}
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py:634]: Just wondering how many times we are expecting this as below you are asserting that it is in the set. You might as well assert equal to test name. Asserting that it is in a set doesn't tell me how many times you are expecting it

File: tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py [hunk 2]
```diff
@@ -602,3 +603,155 @@ def mock_fetch_data_holder(config: IngestDataConfig) -> SQLDataHolder:
...
+        assert len(valid_event_ids) == 0
+
+        # Test 2: ingest_data = True
+        config = self.get_ingest_config(
+            mock_yaml_config_string, mock_temp_dir_with_json_files
+        )
+        result = otel_to_pv(config, ingest_data=True)
+
+        events = []
+        valid_job_names = {"Backend TestJob", "Frontend TestJob"}
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py:660]: Again with a set I can't see how many times you are expecting each job name (I assume two but from the tests I can't tell)

File: tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py [hunk 3]
```diff
@@ -602,3 +603,155 @@ def mock_fetch_data_holder(config: IngestDataConfig) -> SQLDataHolder:
...
+        assert len(valid_event_ids) == 0
+
+        # Test 4: async_flag = True
+        result = otel_to_pv(ingest_data_config, async_flag=True)
+        events = []
+        valid_event_ids = [f"{i}_{j}" for i in range(5) for j in range(2)]
+        valid_job_ids = {f"test_id_{i}" for i in range(5)}
+        job_id_count = {}
+        for job_name, pv_event_streams in result:
+            assert job_name in valid_job_names
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py:721]: same as above

File: tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py [hunk 4]
```diff
@@ -602,3 +603,155 @@ def mock_fetch_data_holder(config: IngestDataConfig) -> SQLDataHolder:
...
+        result = otel_to_pv(
+            ingest_data_config,
+            event_to_async_group_map=event_to_async_group_map,
+        )
+        events = []
+        valid_event_ids = [f"{i}_{j}" for i in range(5) for j in range(2)]
+        valid_job_ids = {f"test_id_{i}" for i in range(5)}
+        job_id_count = {}
+        for job_name, pv_event_streams in result:
+            for pv_event_gen in pv_event_streams:
```
Comment (FreddieMatherSmartDCSIT) [inline tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py:746]: no check for job name here

File: tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py (hunk 1 above)
Comment (AdrianMontaguSmartDCSIT) [reply tests/tel2puml/otel_to_pv/test_sequence_otel_v2.py:634]: Updated

**Review by FreddieMatherSmartDCSIT — APPROVED** [review]
Looks good to me

**Pull request merged**